### ISSUE TITLE
Make check for capabilities optional

### ISSUE THREAD

**Issue by Bables55** [nextcloud/richdocuments#949]
I updated the Collaboraonline app from 3.5.3 to 3.6.0. Then the office server is no longer accessible.
I've downgraded to 3.5.3 and it's still not working. I am currently using the "WOPI" app as an alternative. If this is the case, it is available.

In the case of the Collaboraonline app, no matter what type of content I type in the server address entry field, it shows my previous address.

When I looked at the 3.6.0 configuration items, I found them to be very simplistic. The icon above indicating the state of the connection was red. Of course there is no mistake in the WOPI server address.

No error was output.

3.6.0 has a fatal flaw. Better not to update it.

---
Nextcloud 18.0.4, PHP 7.3.17, loolwsd 4.6.3

**Comment by Tealk** [comment]
I can confirm this, I always get the message that the server is not configured, I have no problems with 3.5.3

**Comment by Bables55** [comment]
I went back to 3.5.3 and I can't even set it to blank.

**Comment by taavit21** [comment]
Can confirm. I have the exact same issue. Had to disable collabora because downgrading to 3.5.3 doesnt work either.

**Comment by Tealk** [comment]
> Can confirm. I have the exact same issue. Had to disable collabora because downgrading to 3.5.3 doesnt work either.

first disable the add-on with version 3.6.0 and then delete it, then you can upload and activate version 3.5.3 via ftp
Download: https://github.com/nextcloud/richdocuments/releases/download/v3.5.3/richdocuments.tar.gz

**Comment by juliushaertl** [comment]
I cannot reproduce any issue with that. Please fill out the complete issue template especially if there is something in the browser console or nextcloud logs. 

> In the case of the Collaboraonline app, no matter what type of content I type in the server address entry field, it shows my previous address.

Maybe you can share a quick recording of the exact behavior there?

**Comment by Tealk** [comment]
I hope someone can take over, I just have the live instance of my company and I don't want to test it there.

**Comment by buttle** [comment]
Same here. Staff during times of covid are collabora dependent. Please fix soon.

**Comment by juliushaertl** [comment]
@buttle Please use the [Github Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) feature instead of commenting if there is no additional information provided.

**Comment by taavit21** [comment]
@juliushaertl 
I think this might be the issue here.

![Capture](https://user-images.githubusercontent.com/47745930/81647560-e25acb00-9435-11ea-8a30-4815d14a6716.PNG)

**Comment by martinjakl** [comment]
Same problem here.
When I change in viewer.js
isCollaboraConfigured:"object"===f(OC.getCapabilities().richdocuments.collabora)&&0!==OC.getCapabilities().richdocuments.collabora.length
to
isCollaboraConfigured:true
it works again

**Comment by juliushaertl** [comment]
@taavit21 Is that the issue when saving? It seems that for some reason it is redirecting to /apps/files/. Are you running Nextcloud with a rewrite so that urls are not containing index.php?

**Comment by thebearon** [comment]
@juliushaertl: it seems like richdocuments stopped accepting Collabora Online if the `hosting/capabilities` endpoint is not available.

For everyone, if you are using a reverse proxy, please check if you have the following (or similar, depending on SSL/termination) configuration entries:
``` 
# Capabilities
ProxyPass           /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities retry=0
ProxyPassReverse    /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities
```
https://www.collaboraoffice.com/code/apache-reverse-proxy/
```
    # Capabilities
    location ^~ /hosting/capabilities {
        proxy_pass https://localhost:9980;
        proxy_set_header Host $http_host;
    }
```
https://www.collaboraoffice.com/code/nginx-reverse-proxy/

Generally, make sure `https://<server:port>/hosting/discovery` of the Collabora Online/CODE instance is accessible either directly, or via reverse proxy.

**Comment by Tealk** [comment]
@thebearon is that new? because it is not included in the nextcloud guide linked from nextcloud
https://icewind.nl/entry/collabora-online/

**Comment by taavit21** [comment]
> @taavit21 Is that the issue when saving? It seems that for some reason it is redirecting to /apps/files/. Are you running Nextcloud with a rewrite so that urls are not containing index.php?

I think it´s irrelevant to the current issue.
I downgraded to 3.5.3 as a temporary solution.

**Comment by kiki78** [comment]
I confirm that downgrade to 3.5.3 is temporary solution for me.
Sorry I can't help more :-(

**Comment by juliushaertl** [comment]
Ok, yes we indeed use the capabilities endpoint now to check if a server is actually reachable properly. 

@taavit21 @kiki78 Please check your webserver setup as described by @thebearon in that case.

**Comment by taavit21** [comment]
> Ok, yes we indeed use the capabilities endpoint now to check if a server is actually reachable properly.
> 
> @taavit21 @kiki78 Please check your webserver setup as described by @thebearon in that case.

I would say there is still a problem because at first check without adding endpoint it says connection to collabora server is successful.

Demonstrated here:

![Capture](https://user-images.githubusercontent.com/47745930/81673113-ca923f80-9453-11ea-8697-8c8f3cca4847.PNG)

**Comment by kiki78** [comment]
Sorry if my response is irrelevant.

From my desktop browser I join https://my_collabora_server/hosting/discovery and receive XML page.
If I try https://my_nextcloud_server:9980/hosting/discover this doesn't work

After, I connect in console to my nextcloud server.
wget https://my_collabora_server/hosting/discovery, receive file
wget https://my_nextcloud_server:9980/hosting/discover don't work

wget localhost:9980/hosting/discovery :  connected.
HTTP request sent, awaiting response... No data received.

wget https://localhost:9980/hosting/discovery :  connected.
ERROR: The certificate of ‘localhost’ is not trusted.

For information Collabora is on same server but in Docker.
We use Nginx and Let's Encrypt for certificats.

With version 3.6.0, I confirm taavit21 : Collabora apps page say reachable when press Save, but If I refresh, I see not reachable.

Regards,

**Comment by weberhofer** [comment]
If the instance is still not working after adding hosting/capabilities have a look at #948

**Comment by thebearon** [comment]
> Ok, yes we indeed use the capabilities endpoint now to check if a server is actually reachable properly.

@juliushaertl, could we keep relying on `hosting/discovery` instead? `hosting/capabilities` isn't necessarily there... or to be more informative, check both, and warn the user if only `hosting/discovery` is available that some features will be missing? (like template handling and previews)

Who could update the sample proxy configuration in the following guide?
https://nextcloud.com/collaboraonline/

**Comment by juliushaertl** [comment]
> could we keep relying on hosting/discovery instead? hosting/capabilities isn't necessarily there... or to be more informative, check both, and warn the user if only hosting/discovery is available that some features will be missing? (like template handling and previews)

Yes, that sounds like a solid plan.

> Who could update the sample proxy configuration in the following guide?
nextcloud.com/collaboraonline

@jospoortvliet can help there I guess. We should definitely just link that to https://www.collaboraoffice.com/code/apache-reverse-proxy/ and https://www.collaboraoffice.com/code/nginx-reverse-proxy/ so we don't have multiple places where this might get forgotten if something changes.

**Comment by Bables55** [comment]
Disable and remove 3.6.0. Then install 3.5.3. I was only able to use 3.5.3 right after that. But after a certain amount of time, it was no longer available again. The user becomes a downloader of the file. I went back to using the "WOPI app" again. It really is dangerous to use 3.6.0.

**Comment by axheli** [comment]
I'm having the same issue since i updated to 3.6.0 once i rolled back to 3.5.3 it worked.

**Comment by doudz** [comment]
I also have the same problem, using nginx and I solved it by adding a missing / new entry in my nginx configuration file, according to https://www.collaboraoffice.com/code/nginx-reverse-proxy/
```
# Capabilities
    location ^~ /hosting/capabilities {
        proxy_pass http://localhost:9980;
        proxy_set_header Host $http_host;
    }
```

The official french doc https://nextcloud.com/fr_FR/collaboraonline/  refers to https://icewind.nl/entry/collabora-online/  where  /hosting/capabilities entry is missing

**Comment by kiki78** [comment]
Finally, I recheck my collabora.conf and only /hosting/discovery is inside, not /hosting.capabilities.
So I add lines like thebaron and doudz and now 3.6.0 seems to work :-)

**Comment by Bables55** [comment]
I'm wondering if something needs to be done after 3.6.0 and it's no longer supported by regular updates.

**Comment by Bables55** [comment]
I wonder if 3.6.0 and later require additional settings and can no longer be supported by regular updates?

**Comment by axheli** [comment]
Yes i was missing 

ProxyPass           /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities retry=0
ProxyPassReverse    /hosting/capabilities https://127.0.0.1:9980/hosting/capabilities

at the end  i updated to 3.6.0 and no issues

**Comment by taavit21** [comment]
> I also have the same problem, using nginx and I solved it by adding a missing / new entry in my nginx configuration file, according to https://www.collaboraoffice.com/code/nginx-reverse-proxy/
> 
> ```
> # Capabilities
>     location ^~ /hosting/capabilities {
>         proxy_pass http://localhost:9980;
>         proxy_set_header Host $http_host;
>     }
> ```
> 
> The official french doc https://nextcloud.com/fr_FR/collaboraonline/ refers to https://icewind.nl/entry/collabora-online/ where `/hosting/capabilities` entry is missing

I can confirm that this resolved issue for me aswell.

**Comment by Thom1b** [comment]
> I also have the same problem, using nginx and I solved it by adding a missing / new entry in my nginx configuration file, according to https://www.collaboraoffice.com/code/nginx-reverse-proxy/
> 
> ```
> # Capabilities
>     location ^~ /hosting/capabilities {
>         proxy_pass http://localhost:9980;
>         proxy_set_header Host $http_host;
>     }
> ```
> 
> The official french doc https://nextcloud.com/fr_FR/collaboraonline/ refers to https://icewind.nl/entry/collabora-online/ where `/hosting/capabilities` entry is missing

Works for me. Thanks!

**Comment by Djiock** [comment]
> I also have the same problem, using nginx and I solved it by adding a missing / new entry in my nginx configuration file, according to https://www.collaboraoffice.com/code/nginx-reverse-proxy/
> 
> ```
> # Capabilities
>     location ^~ /hosting/capabilities {
>         proxy_pass http://localhost:9980;
>         proxy_set_header Host $http_host;
>     }
> ```
> 
> The official french doc https://nextcloud.com/fr_FR/collaboraonline/ refers to https://icewind.nl/entry/collabora-online/ where `/hosting/capabilities` entry is missing

Did the trick for me too, careful when using SSL on both ends like me : 
`proxy_pass https://localhost:9980`

**Comment by Bables55** [comment]
I've added "# Capabilities~~~" to nginx.conf. However, I can't connect to Collabora with the following error.

> Collabora Online is not setup yet.
> Click here to configure your own server or connect to a demo server.

**Comment by juliushaertl** [comment]
> Is this a legitimate solution? Is it a temporary workaround?

Well if your Collabora server is using SSL then you would need to connect using that of course.

**Comment by husisusi** [comment]
This fixed my issue and I can now use 3.6 and 3.7,


    disable app
    remove app folder
    run the following sql statements:
    drop table oc_richdocuments_direct;
    drop table oc_richdocuments_wopi;
    drop table oc_richdocuments_assets;
    delete from oc_migrations where app='richdocuments';
    reinstall the app
Drom: https://github.com/nextcloud/richdocuments/issues/917#issuecomment-619104454

**Comment by raffisweb** [comment]
Hi there, could someone please point me to a solution for traefik?

**Comment by ludogre** [comment]
Hi!
I've got the same problem.

> This fixed my issue and I can now use 3.6 and 3.7,
> 
> ```
> disable app
> remove app folder
> run the following sql statements:
> drop table oc_richdocuments_direct;
> drop table oc_richdocuments_wopi;
> drop table oc_richdocuments_assets;
> delete from oc_migrations where app='richdocuments';
> reinstall the app
> ```
> 
> From: [#917 (comment)](https://github.com/nextcloud/richdocuments/issues/917#issuecomment-619104454)

It didn't work for me.

**Comment by joshtrichards** [comment]
This was worked out long ago, closing. Docs appear accurate today as well - e.g.

https://sdk.collaboraonline.com/docs/installation/Proxy_settings.html#reverse-proxy-with-apache-2-webserver
